pkg/model/model.go

```
@@ -1039,3 +1047,47 @@ func New(
 	m.ApplyShapeIgnoreRules()
 	return m, nil
 }
+
+// AddCustomTypeDefinitions adds type definitions for custom shapes to the typeDefs slice
+func (m *Model) AddCustomTypeDefinitions(tdefs []*TypeDef) ([]*TypeDef, error) {
+	var customTypes []*TypeDef
+	for _, customShape := range m.SDKAPI.CustomShapes {
+		shape := customShape.Shape
+		if shape.Type != "structure" {
+			continue
+		}
+
+		shapeExists := false
+		for _, tdef := range tdefs {
+			if tdef.Shape.ShapeName == shape.ShapeName {
+				shapeExists = true
+				break
+			}
+		}
+		if shapeExists {
+			continue
+		}
```

can we just return an error here?

---

also `shapeExists` variable is not needed

---

Declaration

```
ReplicaCustomStatus:
      RegionName: string
      ReplicaStatus: string
      ReplicaStatusDescription: string
      ReplicaStatusPercentProgress: string
      ReplicaInaccessibleDateTime: string
      
```

Reference
```
        ReplicasDescriptions:
        custom_field:
          list_of: ReplicaCustomStatus
        is_read_only: true
```
  
 when `newList(memberShapeName string) ` is called to create the list in `InjectCustomShapes` it tries to create the same shape again. Need to verify that if its created its not done again.

---

if I panic instead i get `panic: shape ReplicaCustomStatus already exists`

---

Should `custom_field.list_of` and `custom_field.map_of` inject a post fix to the shape name?